### PR TITLE
update windows build image for the agent to a smaller image

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### PR Description

This PR creates a new windows build image for the agent

#### Which issue(s) this PR fixes

Prereq step for #646

#### Notes to the Reviewer

This draft is for gathering feedback on the approach and work so far.

Notes
- The image is buildable and runnable in current state. Example commands below.
  - From the root of the project
    - `docker build -f ./build-image/windows/Dockerfile -t grafana/agent-build-image:latest-windows .`
    - `docker run -itd grafana/agent-build-image:latest-windows`
- I started with `library/golang:1.19.4-windowsservercore-1809` (6gb) for the builder rather than `grafana/ci-build-windows:0.3.4` (17gb) due to the size difference.

Testing
- I built the image using the above command
- I started a container using the above command
- From inside the running container
  - I cloned the grafana/agent project in C:\go\src
  - I successfully ran the command from drone.yml `go test -tags="nodocker,nonetwork" ./...`
#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: build-image/windows/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+RUN mkdir C:\\TDM-GCC-64; \
+    Start-Process 7z -ArgumentList 'e C:\\Windows\\temp\\TDM-GCC-64.exe -oC:\\TDM-GCC-64 -y' -Wait; \
+    Start-Process 7z -ArgumentList 'e C:\\TDM-GCC-64\\*.tar.xz -oC:\\TDM-GCC-64 -y' -Wait; \
+    Start-Process 7z -ArgumentList 'x C:\\TDM-GCC-64\\*.tar -oC:\\TDM-GCC-64 -y' -Wait; \
+    Remove-Item "C:\\TDM-GCC-64\\*" -Include *.tar.xz, *.tar -Force; \
+    setx /M PATH $('C:\TDM-GCC-64\bin;' + $Env:PATH); \
+    Remove-Item -Path C:\\Windows\\temp\\TDM-GCC-64.exe -Force
+
+RUN choco install make --version 4.3 -y
+RUN choco install docker -y
```
[the remark this answers is not in the document]
Comment (rfratto) [reply build-image/windows/Dockerfile:31]: It looks like https://community.chocolatey.org/packages/docker-cli took its place, can we install that instead?

[the remark this answers is not in the document]
Comment (erikbaranowski) [reply build-image/windows/Dockerfile:31]: good eye, I will make the update in a little bit

**Review by rfratto — APPROVED** [review]
Thanks! Sorry to scope creep, but since you updated Go here from 1.19.3 to 1.19.4, can you do the same for the other build-image Dockerfile and note it in the changelog that we upgraded to using Go 1.19.4 for builds?

**Comment by erikbaranowski** [comment]
no prob, I will look at it

**Pull request merged**